### PR TITLE
chore(ci): Build snapshot in PRs

### PR DESCRIPTION
**Pull request by nahsi**

### CODE REVIEW

File: .github/workflows/snapshot.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        uses: fluencelabs/github-actions/generate-snapshot-id@main
+
+  snapshot:
+    needs: version
+    permissions:
+      contents: read
+      id-token: write
+    uses: ./.github/workflows/container.yml
+    with:
+      image: "docker.fluence.dev/distro"
```
Comment (folex) [inline .github/workflows/snapshot.yml:33]: hmmm, maybe `docker.fluence.dev/rust-peer:distro-$BRANCHNAME`? it's kinda strange to have basically the same image in `distro` and `rust-peer` 🤔 

I don't think "source" (from which repo it comes) should matter, only artifact semantics matters.

Comment (nahsi) [reply .github/workflows/snapshot.yml:33]: Well knowing source is good. We only have rust-peer images in our distro :) Maybe something like `docker.fluence.dev/rust-peer-distro/rust-peer:branchetc` is better?

Comment (nahsi) [reply .github/workflows/snapshot.yml:33]: I like to see repo name in image name. It will be easier to clean up later

Comment (folex) [reply .github/workflows/snapshot.yml:33]: no `/distro` is better. well, ok, if you insist :) Let's keep status quo and see how it goes.

**Review by folex — APPROVED** [review]

**Pull request merged**